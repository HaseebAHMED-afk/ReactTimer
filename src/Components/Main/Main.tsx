import React, { useState, useEffect } from 'react'
import CalculateTimer from '../Tools/CalculateTimer';
import Controls from '../Controls/Control';
import '../../App.css'

const Main: React.FC = (): JSX.Element => {
      const [timeInSeconds, setTimeInSeconds] = useState<number>(0);
      const [timerArray, setTimerArray] = useState<Array<number|string>>([]);

      useEffect(() => {
           let timeArray: Array<number|string> = CalculateTimer(timeInSeconds);
           setTimerArray(timeArray);
      },[timeInSeconds])

      return (
            <div>
            <div className="time-container">
                  <h1 className="timer">{timerArray[0]}:{timerArray[1]}:{timerArray[2]}</h1>
            </div>
            <Controls time={setTimeInSeconds}/>
            </div>
      )
}

export default Main