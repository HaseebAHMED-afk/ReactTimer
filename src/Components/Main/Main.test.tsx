import React from 'react';
import {shallow} from 'enzyme';
import Main from './Main';

describe('Main Components Test Case', () => {
    let container : any;

    beforeEach(() => (container = shallow(<Main />)))
    it('renders the main component', ()=> {
        expect(container).toMatchSnapshot()
    })

    it('renders the headings',() => {
        expect(container.find('h1').length).toEqual(1)
    })

    it('renders the divs', () => {
        expect(container.find('div').length).toEqual(2)
    })

    it('renders control components', ()=>{
        expect(container.find('Controls').length).toEqual(1)
    })
})
