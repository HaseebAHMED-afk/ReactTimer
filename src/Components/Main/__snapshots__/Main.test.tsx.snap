// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Components Test Case renders the main component 1`] = `
<div>
  <div
    className="time-container"
  >
    <h1
      className="timer"
    >
      :
      :
    </h1>
  </div>
  <Controls
    time={[Function]}
  />
</div>
`;
